enhancements/ingress/configurable-dns-loadbalancerservice.md

```diff
@@ -0,0 +1,316 @@
+the wildcard *DNSRecord* associated with the controller is managed/unmanaged. 
+
+### Implementation Details/Notes/Constraints
+
+Based on `.loadBalancer.dns`, the controller decides whether to ensure creating
+a wildcard *DNSRecord*. If the customer is updating `.loadBalancer.dns` from 
+`"Managed"` to `"Unmanaged"` the previously created *DNSRecord* will be updated
+appropriately.
+
+The existing state of the ingresscontroller will be synced to the *DNSRecord* as
+well, to clearly state which dns record is unmanaged. The cluster admin can choose
+to delete the *DNSRecord* at his own discretion. 
+
+Once the *DNSRecord* is set to `Unmanaged`, any updates done to it will be ignored
+by the operator, but it to be noted that if the ingresscontroller is updated from
+`Unmanaged` to `Managed` any changes done will be lost during re-sync.
+
+The support provided by the installer and this new field aren't entirely
+mutually exclusive, i.e. if customer has opted to disable DNS management cluster
+wide (as documented
+[here](https://github.com/openshift/installer/blob/master/docs/user/aws/install_upi.md#remove-dns-zones)),
+this field is of no value and need not be set since *DNSRecord* created is silently
+reconciled by the DNS controller.
+
+### Risks and Mitigations
+
+This feature does not support default ingresscontrollers, if updated to be unmanaged
+in conjunction with updating the domain could result in breaking cluster connectivity
+if not all components are properly updated. This should be done at the discretion
+of the cluster admin. The manual creation of the required DNS records will need to be
+done prior to making this change, as the canary controller will try to ensure that the
+ingress domains are reachable, and otherwise this sets the `cluster-ingress-operator`
+in a degraded state.
+
+### Drawbacks
+
+After an ingresscontroller is updated from `Managed` to `Unmanaged`, the previous
+*DNSRecord* is left in an `Unmanaged` state which could result in orphaned resources
+on the cluster and the cloud provider.
+
+## Design Details
+
+### Open Questions
+
+Currently, as per documentation `.spec.domain` in the ingresscontroller cannot be
+updated after creation since updating the [domain](https://github.com/openshift/cluster-ingress-operator/blob/972f09b9dbb181ae5c414da2a990b57c60fde9d8/pkg/operator/controller/ingress/controller.go#L342-L355)
+in the ingresscontroller doesn't result in [updating](https://github.com/openshift/cluster-ingress-operator/blob/972f09b9dbb181ae5c414da2a990b57c60fde9d8/pkg/operator/controller/ingress/dns.go#L97)
+the wildcard *DNSRecord*. 
+> 1. Can this logic be changed to support moving from managed DNS to
+     unmanaged DNS on the `default` ingresscontroller?
+     
+  __Closed:__ The operator will continue to retain immutability of `.spec.domain`
+              and will not support updating domain when migrating from `Managed` to 
+              `Unmanaged` DNS.
+  
+
+### Test Plan
```

Comment:
This may or may not be the right spot for it but the orphaned `DNSRecord` case should be tested as well.